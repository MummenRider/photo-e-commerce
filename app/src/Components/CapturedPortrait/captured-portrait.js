import {
  Container,
  Block,
  Left,
  MainTitle,
  SubTitle,
  Right,
  TextContainer,
} from "Components/CapturedPortrait/captured-portrait.styles";
import { useInView } from "react-intersection-observer";
export default function CapturedPortrait({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}
CapturedPortrait.Block = function CapturedPortraitBlock({
  children,
  ...restProps
}) {
  return <Block {...restProps}>{children}</Block>;
};
CapturedPortrait.Left = function CapturedPortraitLeft({
  children,
  ...restProps
}) {
  return <Left {...restProps}>{children}</Left>;
};
CapturedPortrait.TextContainer = function CapturedPortraitTextContainer({
  options,
  children,
  ...restProps
}) {
  const [ref, inView] = useInView({ ...options });
  return (
    <TextContainer
      animate={{ opacity: inView ? 1 : 0, y: inView ? 0 : 50 }}
      transition={{ duration: 0.6, ease: [0.43, 0.13, 0.23, 0.96] }}
      {...restProps}
      ref={ref}
    >
      {children}
    </TextContainer>
  );
};
CapturedPortrait.SubTitle = function CapturedPortraitSubTitle({
  children,
  ...restProps
}) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};
CapturedPortrait.MainTitle = function CapturedPortraitMainTitle({
  children,
  ...restProps
}) {
  return <MainTitle {...restProps}>{children}</MainTitle>;
};
CapturedPortrait.Right = function CapturedPortraitRight({
  options,
  children,
  ...restProps
}) {
  const [ref, inView] = useInView({ ...options });
  return (
    <Right
      {...restProps}
      style={{ transformOrigin: "top center" }}
      transition={{
        ease: [0.43, 0.13, 0.23, 0.96],
        duration: 0.8,
      }}
      animate={{
        scale: inView ? 1 : 1.2,
      }}
      ref={ref}
    >
      {children}
    </Right>
  );
};
