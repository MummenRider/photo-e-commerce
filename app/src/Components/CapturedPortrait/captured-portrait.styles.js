import styled from "styled-components";
import { motion } from "framer-motion";
export const Container = styled.section`
  color: white;
  display: flex;
  flex-direction: column;
`;
export const Block = styled.div`
  display: flex;
  @media screen and (max-width: 768px) {
    flex-direction: column-reverse;
  }
`;
export const Left = styled.div`
  height: auto;
  width: 100%;
  background-color: ${(props) => props.bkg};
  color: ${(props) => props.fontColor};
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  @media screen and (min-width: 1024px) {
    width: 50%;
  }
`;
export const SubTitle = styled.span`
  font-family: var(--ff-primary);
  font-size: clamp(1.2rem, 1.2vw, 8rem);
  font-weight: 100;
  padding: 4em 0;
`;
export const MainTitle = styled.h1`
  font-family: var(--ff-secondary);
  font-size: clamp(2.2rem, 2.5vw, 16rem);
  font-weight: lighter;
  text-align: center;
  line-height: 1.3;
  letter-spacing: 2px;
  padding: 0 1em;
  width: 80%;
  @media screen and (min-width: 1024px) {
    padding: 0 3em;
    width: 100%;
  }
`;
export const Right = styled(motion.div)`
  height: auto;
  width: 100%;
  @media screen and (min-width: 1024px) {
    width: 50%;
  }
`;
