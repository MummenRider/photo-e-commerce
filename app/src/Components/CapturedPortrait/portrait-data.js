export const capturedImageData = [
  {
    id: 1,
    mainTitle: "Lorem Ipsum Dolor Amet Conseteur",
    topSubTitle: "LOREM, IPSUM, NONUMY",
    bottomSubTitle: "Stet Clita - AMOR",
    src: require("Assets/Images/captured_img_1.jpg").default,
    fallback: require("Assets/Images/captured_img_1.webp").default,
    alt: "Chan image",
    bkg: "#111112",
  },
  {
    id: 2,
    mainTitle: "Lorem Ipsum Dolor Amet Conseteur",
    topSubTitle: "LOREM, IPSUM, NONUMY",
    bottomSubTitle: "Stet Clita - AMOR",
    src: require("Assets/Images/captured_img_2.jpg").default,
    fallback: require("Assets/Images/captured_img_2.webp").default,
    alt: "Ariana image",
    bkg: "#272727",
  },
  {
    id: 3,
    mainTitle: "Lorem Ipsum Dolor Amet Conseteur",
    topSubTitle: "LOREM, IPSUM, NONUMY",
    bottomSubTitle: "Stet Clita - AMOR",
    src: require("Assets/Images/captured_img_3.jpg").default,
    fallback: require("Assets/Images/captured_img_3.webp").default,
    alt: "Liz image",
    bkg: "#D0D5D8",
    fontColor: "#111112",
  },
  {
    id: 4,
    mainTitle: "Lorem Ipsum Dolor Amet Conseteur",
    topSubTitle: "LOREM, IPSUM, NONUMY",
    bottomSubTitle: "Stet Clita - AMOR",
    src: require("Assets/Images/captured_img_4.jpg").default,
    fallback: require("Assets/Images/captured_img_4.webp").default,
    alt: "Bailey image",
    bkg: "#111112",
  },
];
