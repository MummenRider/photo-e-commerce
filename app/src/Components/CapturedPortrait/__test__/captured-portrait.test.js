import { getByAltText, render } from "@testing-library/react";
import CapturedPortrait from "Components/CapturedPortrait/captured-portrait";
import { capturedImageData } from "Components/CapturedPortrait/portrait-data";
import LoadImage from "Components/ImageBlock/image-block";
import { useInView } from "react-intersection-observer";
import { mockAllIsIntersecting } from "react-intersection-observer/test-utils";

describe("<CapturedPortrait />", () => {
  it("should render <CapturedPortrait /> with populated data", () => {
    const { container, getByAltText, getAllByText } = render(
      <CapturedPortrait>
        {capturedImageData.map((imageData) => (
          <CapturedPortrait.Block key={imageData.id}>
            <CapturedPortrait.Left>
              <CapturedPortrait.TextContainer>
                <CapturedPortrait.SubTitle>
                  {imageData.topSubTitle}
                </CapturedPortrait.SubTitle>
                <CapturedPortrait.MainTitle>
                  {imageData.mainTitle}
                </CapturedPortrait.MainTitle>
                <CapturedPortrait.SubTitle>
                  {imageData.bottomSubTitle}
                </CapturedPortrait.SubTitle>
              </CapturedPortrait.TextContainer>
            </CapturedPortrait.Left>
            <CapturedPortrait.Right>
              <LoadImage
                src={imageData.src}
                fallback={imageData.fallback}
                alt={imageData.alt}
              />
            </CapturedPortrait.Right>
          </CapturedPortrait.Block>
        ))}
      </CapturedPortrait>
    );
    mockAllIsIntersecting(true);

    expect(getAllByText("LOREM, IPSUM, NONUMY")).toBeTruthy();
    expect(getAllByText("Lorem Ipsum Dolor Amet Conseteur")).toBeTruthy();
    expect(getAllByText("Stet Clita - AMOR")).toBeTruthy();
    expect(getByAltText("Chan image")).toBeTruthy();
    expect(getByAltText("Ariana image")).toBeTruthy();
    expect(getByAltText("Liz image")).toBeTruthy();
    expect(getByAltText("Bailey image")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
  const CapturedPortraitMock = ({ options }) => {
    const [ref, inView] = useInView({ ...options });
    return <div ref={ref}>{inView.toString()}</div>;
  };
  it("should render element when in view", () => {
    const { getByText } = render(
      <CapturedPortraitMock options={{ threshold: 0.5 }}>
        Hi
      </CapturedPortraitMock>
    );
    mockAllIsIntersecting(0.1);
    expect(getByText("false")).toBeTruthy();

    mockAllIsIntersecting(0.5);
    expect(getByText("true")).toBeTruthy();
  });
});
