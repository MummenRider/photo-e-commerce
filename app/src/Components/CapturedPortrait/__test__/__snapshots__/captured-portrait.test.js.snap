// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapturedPortrait /> should render <CapturedPortrait /> with populated data 1`] = `
<section
  class="sc-bdnxRM CRaxc"
>
  <div
    class="sc-gtsrHT gxepyZ"
  >
    <div
      class="sc-dlnjwi crTKpW"
    >
      <div
        class="sc-hKFxyN dZOQqU"
        style="opacity: 0; transform: none;"
      >
        <span
          class="sc-eCApnc gBNxHu"
        >
          LOREM, IPSUM, NONUMY
        </span>
        <h1
          class="sc-jSFjdj ipxJJc"
        >
          Lorem Ipsum Dolor Amet Conseteur
        </h1>
        <span
          class="sc-eCApnc gBNxHu"
        >
          Stet Clita - AMOR
        </span>
      </div>
    </div>
    <div
      class="sc-gKAaRy bGUpMe"
      style="transform-origin: top center; transform: none;"
    >
      <picture
        class="sc-iCoGMd"
      >
        <source
          class="sc-fujyAs"
          type="image/webp"
        />
        <source
          class="sc-fujyAs"
        />
        <img
          alt="Chan image"
          class="sc-pNWdM heEJZC"
        />
      </picture>
    </div>
  </div>
  <div
    class="sc-gtsrHT gxepyZ"
  >
    <div
      class="sc-dlnjwi crTKpW"
    >
      <div
        class="sc-hKFxyN dZOQqU"
        style="opacity: 0; transform: none;"
      >
        <span
          class="sc-eCApnc gBNxHu"
        >
          LOREM, IPSUM, NONUMY
        </span>
        <h1
          class="sc-jSFjdj ipxJJc"
        >
          Lorem Ipsum Dolor Amet Conseteur
        </h1>
        <span
          class="sc-eCApnc gBNxHu"
        >
          Stet Clita - AMOR
        </span>
      </div>
    </div>
    <div
      class="sc-gKAaRy bGUpMe"
      style="transform-origin: top center; transform: none;"
    >
      <picture
        class="sc-iCoGMd"
      >
        <source
          class="sc-fujyAs"
          type="image/webp"
        />
        <source
          class="sc-fujyAs"
        />
        <img
          alt="Ariana image"
          class="sc-pNWdM heEJZC"
        />
      </picture>
    </div>
  </div>
  <div
    class="sc-gtsrHT gxepyZ"
  >
    <div
      class="sc-dlnjwi crTKpW"
    >
      <div
        class="sc-hKFxyN dZOQqU"
        style="opacity: 0; transform: none;"
      >
        <span
          class="sc-eCApnc gBNxHu"
        >
          LOREM, IPSUM, NONUMY
        </span>
        <h1
          class="sc-jSFjdj ipxJJc"
        >
          Lorem Ipsum Dolor Amet Conseteur
        </h1>
        <span
          class="sc-eCApnc gBNxHu"
        >
          Stet Clita - AMOR
        </span>
      </div>
    </div>
    <div
      class="sc-gKAaRy bGUpMe"
      style="transform-origin: top center; transform: none;"
    >
      <picture
        class="sc-iCoGMd"
      >
        <source
          class="sc-fujyAs"
          type="image/webp"
        />
        <source
          class="sc-fujyAs"
        />
        <img
          alt="Liz image"
          class="sc-pNWdM heEJZC"
        />
      </picture>
    </div>
  </div>
  <div
    class="sc-gtsrHT gxepyZ"
  >
    <div
      class="sc-dlnjwi crTKpW"
    >
      <div
        class="sc-hKFxyN dZOQqU"
        style="opacity: 0; transform: none;"
      >
        <span
          class="sc-eCApnc gBNxHu"
        >
          LOREM, IPSUM, NONUMY
        </span>
        <h1
          class="sc-jSFjdj ipxJJc"
        >
          Lorem Ipsum Dolor Amet Conseteur
        </h1>
        <span
          class="sc-eCApnc gBNxHu"
        >
          Stet Clita - AMOR
        </span>
      </div>
    </div>
    <div
      class="sc-gKAaRy bGUpMe"
      style="transform-origin: top center; transform: none;"
    >
      <picture
        class="sc-iCoGMd"
      >
        <source
          class="sc-fujyAs"
          type="image/webp"
        />
        <source
          class="sc-fujyAs"
        />
        <img
          alt="Bailey image"
          class="sc-pNWdM heEJZC"
        />
      </picture>
    </div>
  </div>
</section>
`;
